docs/library/machine.UART.rst

```diff
@@ -203,11 +178,42 @@ Methods
 
    Availability: rp2, esp32, esp8266, mimxrt, cc3200, stm32, nrf ports, renesas-ra
 
+.. method:: UART.irq(handler=None, trigger=0, hard=False)
+
+   Configure an interrupt handler to be called when a UART event occurs.
+
+   The arguments are:
+
+     - *handler* is an optional function to be called when the interrupt event
+       triggers.  The handler must take exactly one argument which is the
+       ``UART`` instance.
+
+     - *trigger* configures the event which can generate an interrupt.
+       Possible values are:
+
+       - ``UART.IRQ_RXIDLE`` interrupt after receiving at least one character
+         and then the RX line goes idle.
```

Comment:
UART.IRQ_RXIDLE is only supported by the STM32. The CC3200 and Renesas create an interrupt for every received character. Maybe called IRQ_RX, At the CC3200 port, the event is called RX_ANY.